### PR TITLE
viofs: add -F option

### PR DESCRIPTION
**Pull request by EDI-Systems**
Windows has a restriction that the filesystem name must be "NTFS" to run program with admin privilege from it.

Add -F option and allow viofs to masquerade itself as any filesystem (e.g. NTFS) so the Windows restriction is lifted and we may run lots of useful tools and perform software installs from it.

WinFsp's devels are aware of it and in their example memfs they did it this way. The following is an excerpt from their FAQ.

https://winfsp.dev/doc/Frequently-Asked-Questions/

2. I cannot run a program located in a WinFsp drive as administrator. I cannot run regedit.exe from within a WinFsp drive.

When running an executable as administrator, the Windows OS seems to require that the name of the file system that is housing the executable is "NTFS". For example, the MEMFS file system with the command line memfs-x64.exe -i -F NTFS -m X: works.

### CODE REVIEW

**Comment by EDI-Systems** [comment]
The -F option is optional. When not designated it falls back to the default VirtIO-FS.

**Comment by viktor-prutyanov** [comment]
> Windows OS seems to require that the name of the file system that is housing the executable is "NTFS"

But how executables are running from FAT32?

**Comment by EDI-Systems** [comment]
> > Windows OS seems to require that the name of the file system that is housing the executable is "NTFS"
> 
> But how executables are running from FAT32?

NTFS and FAT32 (and even samba-mounted anything), all these windows built-ins are fine. but others are not allowed to pass this check, I guess cause windows haven't seen them. WinFsp already pointed this out.

**Comment by EDI-Systems** [comment]
> > > > Windows OS seems to require that the name of the file system that is housing the executable is "NTFS"
> > > 
> > > 
> > > But how executables are running from FAT32?
> > 
> > 
> > NTFS and FAT32 (and exFAT, or even SAMBA-mounted anything), all these Windows built-in filesystems are fine. but others are not allowed to pass this check, I guess cause they are alien to Windows. WinFsp devs already pointed this out. I tried and the original doesn't allow (as it gives VirtIO-FS filesystem name), but this modified version I'm currently using (which assimilates the WinFsp example MEMFS's solution) allows to run exe with admin.
> 
> So, if you pass -F FAT32, does it also work?

Yes it works, I can run AS SSD benchmark / CPU-Z / EVGA OC installer, etc. from it.

**Comment by EDI-Systems** [comment]
> > > > > Windows OS seems to require that the name of the file system that is housing the executable is "NTFS"
> > > > 
> > > > 
> > > > But how executables are running from FAT32?
> > > 
> > > 
> > > NTFS and FAT32 (and exFAT, or even SAMBA-mounted anything), all these Windows built-in filesystems are fine. but others are not allowed to pass this check, I guess cause they are alien to Windows. WinFsp devs already pointed this out. I tried and the original doesn't allow (as it gives VirtIO-FS filesystem name), but this modified version I'm currently using (which assimilates the WinFsp example MEMFS's solution) allows to run exe with admin.
> > 
> > 
> > So, if you pass -F FAT32, does it also work?
> 
> Yes it works, I can run AS SSD benchmark / CPU-Z / EVGA OC installer, etc. from it.

Now I have a 24TB "FAT32" disk.............

**Comment by YanVugenfirer** [comment]
Hi @EDI-Systems , please add "sign-off" according to the contribution guidelines with your real name: https://github.com/virtio-win/kvm-guest-drivers-windows/blob/master/contributing.md

Optional: if you can comment on your use case here: https://github.com/virtio-win/kvm-guest-drivers-windows/discussions/726

Thanks,
Yan.

**Comment by EDI-Systems** [comment]
> Hi @EDI-Systems , please add "sign-off" according to the contribution guidelines with your real name: https://github.com/virtio-win/kvm-guest-drivers-windows/blob/master/contributing.md
> 
> Optional: if you can comment on your use case here: #726
> 
> Thanks, Yan.

Done.

**Review by YanVugenfirer — APPROVED** [review]

File: viofs/svc/virtiofs.cpp
Comment (viktor-prutyanov) [inline viofs/svc/virtiofs.cpp:140]: Trailing whitespace:
```suggestion
    VIRTFS(ULONG DebugFlags, bool CaseInsensitive,
```

File: viofs/svc/virtiofs.cpp
Comment (viktor-prutyanov) [inline viofs/svc/virtiofs.cpp:2500]: Trailing whitespace:
```suggestion
        sizeof(VolumeParams.FileSystemName) / sizeof(WCHAR),
```

File: viofs/svc/virtiofs.cpp
Comment (viktor-prutyanov) [inline viofs/svc/virtiofs.cpp:2605]:
```suggestion
        "    -F FileSystemName   [file system name for OS]\n"
```

**Review by viktor-prutyanov — CHANGES_REQUESTED** [review]
Hi @EDI-Systems 

I suggest you to add a couple of minor changes and then we can merge the PR.

**Comment by viktor-prutyanov** [comment]
Hi @EDI-Systems 

Could you please squash all the commits into the first one?

**Comment by EDI-Systems** [comment]
> Hi @EDI-Systems
> 
> Could you please squash all the commits into the first one?

squashed and pushed.

**Review by viktor-prutyanov — APPROVED** [review]

**Review by YanVugenfirer — APPROVED** [review]

**Comment by YanVugenfirer** [comment]
@EDI-Systems Thank you for the PR, I am waiting for CI to finish before merging. We have a large queue due to other PRs that touched virtio library code.

**Comment by YanVugenfirer** [comment]
@EDI-Systems Thanks a lot! Merging.

@viktor-prutyanov Please update the WiKi.

**Comment by viktor-prutyanov** [comment]
> Please update the WiKi.

@YanVugenfirer Updated.

**Pull request merged**
